### PR TITLE
tag-enis

### PR DESCRIPTION
**Pull request by calvix**
towards https://github.com/giantswarm/giantswarm/issues/18362

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by calvix** [comment]
sorry that request for review was accidental this is still WIP as I need to finish testing

File: service/internal/cloudconfig/tccpn.go [hunk 1]
```diff
@@ -412,3 +438,31 @@ func (t *TCCPN) newTemplate(ctx context.Context, obj interface{}, mapping hamast
...
+	tags := map[string]string{}
+	for k, v := range labels {
+		if isCloudTagKey(k) {
+			tags[trimCloudTagKey(k)] = v
+		}
+	}
+	fmt.Printf("map: %#v\n", tags)
+
+	t, err := json.Marshal(tags)
+	fmt.Printf("marshaled: %s\n", string(t))
```
Comment (njuettner) [inline service/internal/cloudconfig/tccpn.go:452]: maybe this is a leftover?

File: service/internal/cloudconfig/tccpn.go [hunk 2]
```diff
@@ -412,3 +438,31 @@ func (t *TCCPN) newTemplate(ctx context.Context, obj interface{}, mapping hamast
...
 }
+
+func getCloudTags(labels map[string]string) (string, error) {
+	tags := map[string]string{}
+	for k, v := range labels {
+		if isCloudTagKey(k) {
+			tags[trimCloudTagKey(k)] = v
+		}
+	}
+	fmt.Printf("map: %#v\n", tags)
```
Comment (njuettner) [inline service/internal/cloudconfig/tccpn.go:449]: maybe this is a leftover?

File: service/internal/cloudconfig/tccpn.go [hunk 3]
```diff
@@ -21,6 +24,8 @@ import (
 	"github.com/giantswarm/aws-operator/service/internal/hamaster"
 )
 
+const keyCloudPrefix = "tag.provider.giantswarm.io/"
```
Comment (njuettner) [inline service/internal/cloudconfig/tccpn.go:27]: Shall we move this to https://github.com/giantswarm/aws-operator/blob/master/service/controller/key/common.go#L18? WDYT?

**Review by njuettner — APPROVED** [review]
just minor things, lgtm

**Comment by njuettner** [comment]
Saw the comment a bit late, apologies 🙂

**Comment by calvix** [comment]
ok ready to merge

File: service/internal/cloudconfig/tccpn.go [hunk 4]
```diff
@@ -412,3 +436,29 @@ func (t *TCCPN) newTemplate(ctx context.Context, obj interface{}, mapping hamast
...
+
+	t, err := json.Marshal(tags)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	return string(t), nil
+}
+
+// IsCloudTagKey check is a tag with proper prefix
```
Comment (anvddriesch) [inline service/internal/cloudconfig/tccpn.go:456]:
```suggestion
// IsCloudTagKey checks if a tag has proper prefix
```

File: service/internal/cloudconfig/tccpn.go [hunk 5]
```diff
@@ -412,3 +436,29 @@ func (t *TCCPN) newTemplate(ctx context.Context, obj interface{}, mapping hamast
...
+func isCloudTagKey(tagKey string) bool {
+	return strings.HasPrefix(tagKey, key.KeyCloudPrefix)
+}
+
+// TrimCloudTagKey check is a tag with proper prefix
```
Comment (anvddriesch) [inline service/internal/cloudconfig/tccpn.go:461]:
```suggestion
// TrimCloudTagKey trims key cloud prefix from a tag
```

**Review by anvddriesch — APPROVED** [review]

**Pull request merged**